[build.webkit.org] Bring up open source visionOS builders and testers

robert-jenner opened the pull request:
#### b4e487b4319692a533e7b43a406530fda43c3018
<pre>
[build.webkit.org] Bring up open source visionOS builders and testers
<a href="https://bugs.webkit.org/show_bug.cgi?id=274272">https://bugs.webkit.org/show_bug.cgi?id=274272</a>
<a href="https://rdar.apple.com/121395049">rdar://121395049</a>

Reviewed by Ryan Haddad.

Bring up of OpenSource visionOS builders and testers.

* Tools/CISupport/build-webkit-org/config.json:
* Tools/CISupport/build-webkit-org/factories.py:
(TestFactory.__init__):
* Tools/CISupport/build-webkit-org/factories_unittest.py:
(TestExpectedBuildSteps):
* Tools/CISupport/build-webkit-org/loadConfig.py:
(loadBuilderConfig):
* Tools/CISupport/build-webkit-org/public_html/dashboard/Scripts/Dashboard.js:
* Tools/CISupport/build-webkit-org/public_html/dashboard/Scripts/WebKitBuildbot.js:
(WebKitBuildbot):
* Tools/CISupport/build-webkit-org/public_html/dashboard/Styles/Main.css:
(table.queue-grid tr.platform.visionos-simulator-1 img.logo):
(table.queue-grid tr.platform.visionos-1 img.logo):
* Tools/CISupport/build-webkit-org/steps.py:
(CustomFlagsMixin.customBuildFlag):
(CustomFlagsMixin.appendCustomBuildFlags):
(CustomFlagsMixin.appendCustomTestingFlags):
(CompileWebKit):
(CompileWebKit.evaluateCommand):
(PrintConfiguration.run):
* Tools/CISupport/build-webkit-org/wkbuild.py:
(_should_file_trigger_build):
* Tools/CISupport/build-webkit-org/wkbuild_unittest.py:
(ShouldBuildTest):
(ShouldBuildTest.test_should_build):

Canonical link: <a href="https://commits.webkit.org/279249@main">https://commits.webkit.org/279249@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/afeb514f4771d838cb49b493c039b7e7d1973bd1

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/52816 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/32153 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/5303 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/56095 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/3539 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/55121 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/38870 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/3264 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/42872 "Passed tests") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/2293 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/54914 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/29876 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/45595 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/23987 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/26959 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/2912 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/1698 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/48834 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/3063 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/57688 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/27957 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/3046 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/50265 "Passed tests") | 
| [✅ 🧪 services](https://ews-build.webkit.org/#/builders/28/builds/52822 "Passed tests") | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/29177 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/45787 "Passed tests") | [❌ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/49544 "Found 1 new API test failure: /WebKitGTK/TestWebKitWebContext:/webkit/WebKitWebContext/memory-pressure (failure)") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/11554 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/30096 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/28931 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

aj062 commented on Tools/CISupport/build-webkit-org/factories.py:
```diff
@@ -100,7 +100,7 @@ def __init__(self, platform, configuration, architectures, additionalArguments=N
             self.addStep(InstallWinCairoDependencies())
 
-        if platform.startswith('mac') or platform.startswith('ios-simulator'):
+        if platform.startswith('mac') or platform.startswith('ios-simulator') or platform.startswith('visionos-simulator'):
```
can shorten these ones to something like:

`if platform.startswith(('mac', 'ios-simulator', 'visionos-simulator')):`

ryanhaddad commented on Tools/CISupport/build-webkit-org/factories.py:
```diff
@@ -115,7 +115,7 @@ def __init__(self, platform, configuration, architectures, additionalArguments=N
             self.addStep(SetPermissions())
 
-        if platform.startswith('win') or platform.startswith('mac') or platform.startswith('ios-simulator'):
+        if platform.startswith(('win', 'mac', 'ios-simulator', 'visionos-simulator')):
```
It might be worth removing this one for now, I don't know the state of API tests for visionOS simulator, and I'm not even sure that we've tried running them at all.

robert-jenner replied:
I removed it in both places.

ryanhaddad reviewed (approved):
r=me with comment. Would be good to test this on UAT to make sure we didn't miss anything.

robert-jenner committed:
[build.webkit.org] Bring up open source visionOS builders and testers
https://bugs.webkit.org/show_bug.cgi?id=274272
rdar://121395049

Reviewed by Ryan Haddad.

Bring up of OpenSource visionOS builders and testers.

* Tools/CISupport/build-webkit-org/config.json:
* Tools/CISupport/build-webkit-org/factories.py:
(TestFactory.__init__):
* Tools/CISupport/build-webkit-org/factories_unittest.py:
(TestExpectedBuildSteps):
* Tools/CISupport/build-webkit-org/loadConfig.py:
(loadBuilderConfig):
* Tools/CISupport/build-webkit-org/public_html/dashboard/Scripts/Dashboard.js:
* Tools/CISupport/build-webkit-org/public_html/dashboard/Scripts/WebKitBuildbot.js:
(WebKitBuildbot):
* Tools/CISupport/build-webkit-org/public_html/dashboard/Styles/Main.css:
(table.queue-grid tr.platform.visionos-simulator-1 img.logo):
(table.queue-grid tr.platform.visionos-1 img.logo):
* Tools/CISupport/build-webkit-org/steps.py:
(CustomFlagsMixin.customBuildFlag):
(CustomFlagsMixin.appendCustomBuildFlags):
(CustomFlagsMixin.appendCustomTestingFlags):
(CompileWebKit):
(CompileWebKit.evaluateCommand):
(PrintConfiguration.run):
* Tools/CISupport/build-webkit-org/wkbuild.py:
(_should_file_trigger_build):
* Tools/CISupport/build-webkit-org/wkbuild_unittest.py:
(ShouldBuildTest):
(ShouldBuildTest.test_should_build):

Canonical link: https://commits.webkit.org/279249@main
